### DISCUSSION TITLE
contentNodes(where: {in: "..."}) extremely slow?

### DISCUSSION THREAD

**Discussion by alancwoo** [wp-graphql/wp-graphql#3074] [Q&A]
I would like to fetch any number of items just by their wordpress IDs without knowing their types when querying and hoped to just be able to use `contentNodes(where: {in: "..."})` but it is incredibly slow, even with just a single `ID` (5-6 seconds), even with just a single `ID`.

`get_posts` and `post__in` resolves basically instantaneously.

I was wondering if others are observing this behaviour or if it's something localized?

I am using `Wordpress 6.4.3`, `WPGraphQL 1.22.1` and `WPGraphQL for ACF 2.2.0`

**Comment by justlevine** [comment]
@alancwoo 

If you're looking for a quick fix try WPGraphQL Smart Cache.

If you're looking for some actual debugging, you're gonna have to provide a bit more information, like the full query and the comparative `get_posts()` query, even better if you enable GraphQL debugging and tracing in the settings and share the response, which will include the timings for the individual database calls.

Couple generic notes:
- `contentNodes` fetches across _all_ posts types by default. `get_posts()` only does that if you set `$args['post_type']`.
- `get_posts()` only fetches the `WP_Post` objects and no meta fields. The power of GraphQL is that you can fetch the associated data objects (meta fields - you said you're using ACF, terms, authors, etc) but database joins can get expensive quickly, so you want to make sure you're smart about the query shape. If you want a real comparative benchmark, it's not enough to compare `get_posts()` but all the followup queries for the associated data.
